src/analysis/analysis-model.js

```diff
@@ -123,10 +127,15 @@ module.exports = Model.extend({
   toJSON: function () {
     var json = _.pick(this.attributes, 'id', 'type');
     json.params = _.pick(this.attributes, this.getParamNames());
-    _.each(this._getSourceNames(), function (sourceName) {
+    var sourceNames = this._getSourceNames();
+    _.each(sourceNames, function (sourceName) {
       var source = {};
-      source[sourceName] = this.get(sourceName).toJSON();
-      _.extend(json.params, source);
+      var isOptional = this._camshaftReference.isSourceNameOptionalForAnalysisType(this.get('type'), sourceName);
+      var sourceInfo = this.get(sourceName);
+      if (sourceInfo || !isOptional) {
+        source[sourceName] = sourceInfo.toJSON();
```

Comment:
Same here, I'd only invoke `sourceInfo.toJSON()` if there's a `sourceInfo`. If `sourceInfo` is not defined but it's optional we can just set it to `null` / `undefined` (or whatever the Maps API is waiting for in those cases...):

```
if (sourceInfo) {
  source[sourceName] = sourceInfo.toJSON();
} else {
  if (!isOptional) {
    source[sourceName] = null;  
  }
}
```
```